### PR TITLE
ci: Bump PySide6 version in Fedora end-user envs

### PR DESCRIPTION
**Pull request by apyrgio**
Our end-user Fedora environments, that we create for testing how Dangerzone would operate on a clean Fedora system, require PySide6 to be installed. This package is not available from the official Fedora repos yet.

We have a way instead to check the poetry.lock file, grab the latest PySide6 version from there, and install it from a URL. This is no longer necessary, now that PySide6 6.7.0 will soon be available in all stable Fedora releases. Since the last release maintained by FPF will be 6.6.3.1, we should pin this version in our env.py script. This way, we can bump poetry.lock independently, and let Windows/macOS users get different versions.

Refs freedomofpress/maint-dangerzone-pyside6#5

### CODE REVIEW

File: dev_scripts/env.py
Comment (apyrgio) [inline dev_scripts/env.py:24]: This line has been added to make the GitHub Actions use the packages in https://github.com/freedomofpress/yum-tools-prod/pull/20, and should be removed before merging this PR.

File: dev_scripts/env.py
Comment (legoktm) [inline dev_scripts/env.py:325]: This can be removed now, I think

Comment (apyrgio) [reply dev_scripts/env.py:325]: ... and I should use a proper docstring. Thanks for catching this, I have a fixup in ce39cfe071afc0e693040f03e316e69bb8e9b6b5

Comment (legoktm) [reply dev_scripts/env.py:325]: Your fixup seems to have ended up on https://github.com/freedomofpress/dangerzone/pull/779 instead of this PR, but it LGTM.

**Review by legoktm — APPROVED** [review]
LGTM, modulo the one part that needs to be reverted and the fixup commit associated with the other PR

File: dev_scripts/env.py
Comment (apyrgio) [reply dev_scripts/env.py:325]: Ouch. I will properly squash the commits, thanks for flagging this.

File: dev_scripts/env.py
Comment (apyrgio) [reply dev_scripts/env.py]: This is no longer the case.

**Pull request merged**
